polar-core/src/data_filtering.rs

```diff
@@ -0,0 +1,799 @@
+            if let Value::Expression(op) = rhs.value() {
+                rhs = process_exp(var_info, op).unwrap();
+            };
+
+            match (lhs.value(), rhs.value()) {
+                // l in r
+                (Value::Variable(l), Value::Variable(r)) => {
+                    var_info.in_relationships.push((l.clone(), r.clone()));
+                }
+                // var in [1, 2, 3]
+                (Value::Variable(_var), _val) => {
+                    // @Q(steve): Does this ever actually come through the simplifier?
+                    // @Note(steve): MikeD wishes this came through as an in instead of or-expanded.
+                    // That way we could turn it into an `in` in sql.
+                    unimplemented!("var in list of values constraints are not yet supported for data filtering.");
+                    // var_info.in_values.push((var.clone(), Term::new_temporary(val.clone())));
+                }
+                // 123 in var
+                (val, Value::Variable(var)) => {
+                    var_info
+                        .contained_values
+                        .push((Term::new_temporary(val.clone()), var.clone()));
+                }
+                (_a, _b) => {
+                    // @NOTE: This is probably just a bug if we hit it. Shouldn't get any other `in` cases.
+                    unimplemented!(
+                        "Unknown `in` constraint that is not yet supported for data filtering."
+                    );
+                }
+            };
+        }
+        Operator::Debug => unimplemented!("debug() is not supported for data filtering."),
+        Operator::Print => (),
+        Operator::Cut => unimplemented!("`cut` is not supported for data filtering."),
+        Operator::New => panic!("`new` operation in expression"),
+        Operator::Not => panic!("`not` operation in expression"),
+        Operator::Mul => unimplemented!("multiplication is not supported for data filtering."),
+        Operator::Div => unimplemented!("division is not supported for data filtering."),
+        Operator::Mod => unimplemented!("`mod` is not supported for data filtering."),
+        Operator::Rem => unimplemented!("`rem` is not supported for data filtering."),
+        Operator::Add => unimplemented!("addition is not supported for data filtering."),
+        Operator::Sub => unimplemented!("subtraction is not supported for data filtering."),
+        Operator::Geq => unimplemented!("`>=` is not supported for data filtering."),
+        Operator::Leq => unimplemented!("`<=` is not supported for data filtering."),
+        Operator::Neq => unimplemented!("`!=` is not supported for data filtering."),
+        Operator::Gt => unimplemented!("`>` is not supported for data filtering."),
+        Operator::Lt => unimplemented!("`<` is not supported for data filtering."),
+        // @TODO(steve): Expand or expressions to multiple bindings in the simplifier.
```

Comment:
I think a better way to accomplish this would be in the inverter. Then we can drop the Or operator completely from partials.

Reply:
Yeah, even better. Will make a task, that's a followup thing.